crates/resolver/src/name_server/name_server_pool.rs

```diff
@@ -240,8 +302,19 @@ impl PoolContext {
             options,
             tls,
             opportunistic_encryption,
+            answer_address_filter: None,
         }
     }
+
+    /// Add an answer address filter
+    pub fn set_answer_filter(&mut self, answer_filter: Arc<AccessControlSet>) {
```

Comment:
Suggest this could be a builder-style method, as in #3318, like `fn with_answer_filter(mut self, filter: AccessControlSet) -> Self`.